src/raft.rs

```diff
@@ -395,6 +395,75 @@ impl<T: Storage> Raft<T> {
         self.batch_append = batch_append;
     }
 
+    /// When committing logs, only logs replicated to at least two different
+    /// groups are committed.
+    ///
+    /// The tuple is (`peer_id`, `group_id`). `group_id` should be larger than 0.
+    pub fn assign_commit_groups(&mut self, ids: &[(u64, u64)]) {
+        for (peer_id, group_id) in ids {
+            assert!(*group_id > 0);
+            if let Some(pr) = self.mut_prs().get_mut(*peer_id) {
+                pr.commit_group_id = *group_id;
+            } else {
+                return;
```

Comment:
Is it valid that the `peer_id` in `ids` can not exist in this raft group?

Reply:
Yes, but here should be continue instead of returning.